### PR TITLE
Okta user investigation - Enhancement

### PR DESCRIPTION
**Pull request by OmriItzhak**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-8207)

## Description
New output was added for login country (the country from which the user logged in).

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Okta/Playbooks/playbook-Okta_-_User_Investigation.yml [hunk 1]
```diff
@@ -1700,6 +1741,9 @@ outputs:
 - contextPath: NumOfFailedLogonASN
   description: Number of failed login from ASN by all users.
   type: unknown
+- contextPath: LogonCountry
+  description: The countries from which the user logged in
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Playbooks/playbook-Okta_-_User_Investigation.yml:1745]:
```suggestion
  description: The countries from which the user logged in.
```

File: Packs/Okta/Playbooks/playbook-Okta_-_User_Investigation_README.md [hunk 2]
```diff
@@ -40,7 +47,10 @@ This playbook does not use any sub-playbooks.
 | UserApplication |  Applications used by the user. | unknown |
 | NumOfOktaFailedLogon | Number of failed login. | unknown |
 | NumOfFailedLogonASN | Number of failed login from ASN by all users. | unknown |
+| LogonCountry | The countries from which the user logged in | unknown |
```
Comment (ShirleyDenkberg) [inline Packs/Okta/Playbooks/playbook-Okta_-_User_Investigation_README.md:50]:
```suggestion
| LogonCountry | The countries from which the user logged in. | unknown |
```

File: Packs/Okta/ReleaseNotes/3_1_29.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+
+#### Playbooks
+
+##### Okta - User Investigation
+
+Added new outputs *LogonCountry*.
```
Comment (ShirleyDenkberg) [inline Packs/Okta/ReleaseNotes/3_1_29.md:6]:
```suggestion
Added new output *LogonCountry*.
```

**Comment by ShirleyDenkberg** [comment]
@idovandijk Doc review completed.

**Review by idovandijk — COMMENTED** [review]
Great playbook!

I think I found a logical issue.
In this task:
image

Let's say the input is "United States", and the logs will show that the user also logged in from "Canada", "France" and "United States".

In that case we will decide that the user is working from a permanent country just because "United States" is in the list of countries in the logs.

Maybe it would be better to use the `LogonCountry` key that we created before, with a **"Unique"** transformer, and then check if it equals to the input country.

So if our logs show `"United States", "United States", "United States"`, our condition will compare "United States" to "United States" which means it's a permanent country.

However if the logs will show `"Canada", "United States", "United States"`, then the condition will compare "United States" to `"Canada", "United States"`, and it will determine that the country is not permanent.

What do you think?

Another point is - I would rename LogonCountry to LogonCountries since it can be a list of countries

**Pull request merged**